### PR TITLE
Adding Carnegie Supernova Project I Data Release 3

### PR DESCRIPTION
**Pull request by ado8**
# Adds the Carnegie Supernova Project I Type Ia Supernova Data Release 3 to Astropile 
Collaboration with @tom-hehir @mb010 @erinhay @David-Chemaly @benboyd97 @mattgrayling and others.

Adding a multiband time-series dataset of Type Ia supernovae from CSP DR3. This code is largely based on the yse code in https://github.com/AstroPile/AstroPile_prototype/pull/46, except for the way data is loaded from files (SNooPy format vs SNANA format).

## Link for reference
[CSP Data](https://csp.obs.carnegiescience.edu/data)
- [Krisciunas et al., 2017](https://ui.adsabs.harvard.edu/abs/2017AJ....154..211K/abstract)
- [CSP DR3](https://csp.obs.carnegiescience.edu/data/CSP_Photometry_DR3.tgz)

## HuggingFace Dataset Format
Each example contains:

  - `object_id`: Dataset specific identifier
  - `spec_class`: Spectral class
  - `ra`, `dec`: right ascension, declination
  - `redshift`: redshift of target. : As decided by ordered availability of: `['REDSHIFT_FINAL', 'REDSHIFT_CMB', 'REDSHIFT_HELIO']`
  - `lightcurve`: object with:
                - `band`: band of observation, arr(string),
                - `time`: timestamp of observation in MJD, arr(float)
                - `mag`: Magnitude (CSP natural system) value of observation, arr(float),
                - `mag_err`: Magnitude (CSP natural system) uncertainty of observation, arr(float)

### CODE REVIEW

**Comment by ado8** [comment]
The Carnegie Supernova Project's [site](https://csp.obs.carnegiescience.edu) seems down at the moment (19/4/2024).
test.sh completed successfully before pushing, but now fails without any change in code.

File: download_data.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import os
+import argparse
+import requests
+import tarfile
+import requests
+import os
+def download_text_file(url, file_path):
```
Comment (mb010) [inline download_data.py:7]:
```suggestion
import os

def download_text_file(url, file_path):
```

File: scripts/.DS_Store
Comment (mb010) [inline scripts/.DS_Store:1]: This file should be removed.

File: download_data.py (hunk 1 above)
Comment (mb010) [inline download_data.py:7]:
```suggestion
import os

def download_text_file(url, file_path):
```

File: scripts/csp/csp.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    adsnote = {Provided by the SAO/NASA Astrophysics Data System}
+}
+
+"""
+
+_DESCRIPTION = """\
+Time-series dataset from the Carnegie Supernova Project I Data Release 3 (CSP-I DR3).
+"""
+
+_HOMEPAGE = "https://csp.obs.carnegiescience.edu/data/CSP_Photometry_DR3.tgz"
```
Comment (mb010) [inline scripts/csp/csp.py:48]: Link to the science project rather than the data download.

**Comment by ado8** [comment]
I messed up my branches and put some CfA edits in during commit 63f6591 and 65ae8b4.
I don't usually work with branches, so I don't know how to scrub these or move them to the CfA branch.

Also, the root level download_data.py file looks like it was from a foundation branch I copied things from.
I don't think it should be there, so I deleted it in this branch.

File: scripts/csp/csp.py (hunk 2 above)
Comment (mb010) [reply scripts/csp/csp.py]: Fixed.

File: scripts/cfa/cfa_SECCSN.py [hunk 3]
```diff
@@ -46,7 +46,7 @@
 
 _HOMEPAGE = "https://lweb.cfa.harvard.edu/supernova/lc.standardsystem.sesn_allphot.dat"
```
Comment (mb010) [inline scripts/cfa/cfa_SECCSN.py:47]: This should be the homepage, not the data origin.

File: scripts/cfa/cfa4.py [hunk 4]
```diff
@@ -46,7 +46,7 @@
 
 _HOMEPAGE = "https://lweb.cfa.harvard.edu/supernova/CfA4/cfa4.lc.stdsystem.fi.ascii"
```
Comment (mb010) [inline scripts/cfa/cfa4.py:47]: This should be the homepage, not the data origin.

File: scripts/cfa/cfa_SECCSN.py [hunk 5]
```diff
@@ -46,7 +46,7 @@
...
 
-_LICENSE = ""
+_LICENSE = "CC BY-NC-SA 4.0"
```
Comment (mb010) [inline scripts/cfa/cfa_SECCSN.py:49]: This is confirmed?

**Review by mb010 — CHANGES_REQUESTED** [review]
This uncovered more links to data rather than homepages. I think its fine otherwise. This is supposed to be an accessible review / contribution process, and your changes just make things better. 🙂 No worries on that front.

File: scripts/cfa/cfa_SECCSN.py (hunk 3 above)
Comment (mb010) [reply scripts/cfa/cfa_SECCSN.py:47]:
```suggestion
_HOMEPAGE = "https://lweb.cfa.harvard.edu/supernova/"
```

**Comment by EiffL** [comment]
if you update from the main branch it should work

**Review by mb010 — APPROVED** [review]

**Pull request merged**